### PR TITLE
Sort milestones by closest due date

### PR DESCRIPTION
**Pull request by smeijer**
The default sorting of milestones in github is pretty frustrating. Milestones are used to manage releases; where as the closest one is often more important than the last updated one.

This PR fixes the default sorting, so they are sorted by closest due date, instead of recently updated.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Clicking the "Milestones" button in https://github.com/sindresorhus/got/issues?q=is:open gives me a 404.

**Comment by sindresorhus** [comment]
Can you add it to https://github.com/sindresorhus/refined-github#highlights ? No screenshot needed.

**Comment by bfred-it** [comment]
jQuery returns an array even if it's not found so `if array` will always be true. Refer to #454 for the suggested implementation.

File: extension/content.js [hunk 1]
```diff
@@ -430,6 +430,13 @@ function addTitleToEmojis() {
 	}
 }
 
+function sortMileStonesByClosestDueDate() {
+	const button = $('.subnav-item[href$="milestones"');
```
Comment (bfred-it) [inline extension/content.js]: Missing `]` in selector

File: extension/content.js [hunk 2]
```diff
@@ -430,6 +430,13 @@ function addTitleToEmojis() {
...
+	if (button) {
```
Comment (bfred-it) [inline extension/content.js]: jQuery returns an array even if it's not found so if array will always be true. Refer to #454 for the suggested implementation.

**Comment by hkdobrev** [comment]
@smeijer Do you think you'd be able to pick this up? Thank you!

**Comment by smeijer** [comment]
Sure; I'll fix this asap.

**Comment by smeijer** [comment]
@zackschuster , 
- [x] rebased on master
- [x] implemented a native js solution as requested by @bfred-it : https://github.com/sindresorhus/refined-github/pull/457#discussion_r119766742
- [x] added to highlights as requested by @sindresorhus : https://github.com/sindresorhus/refined-github/pull/457#issuecomment-305576850

**Comment by sindresorhus** [comment]
It doesn't work if you click the `Milestone` button in https://github.com/chalk/chalk/labels and https://github.com/chalk/chalk/milestone/1

**Comment by smeijer** [comment]
@sindresorhus , I added some page detection's to fix this. It does work for me now, but I have no idea if this is fits in the projects standards. Please take this PR over or close it if it still needs some work.

File: src/content.js [hunk 3]
```diff
@@ -594,6 +602,11 @@ function init(options) {
 
 			if (pageDetect.isMilestone()) {
 				addMilestoneNavigation();
+				sortMileStonesByClosestDueDate();
+			}
+
+			if (pageDetect.isMilestoneList() || pageDetect.isLabelList() || pageDetect.isLabel()) {
+				sortMileStonesByClosestDueDate();
```
Comment (sindresorhus) [inline src/content.js]: Better to have one if statement with all of these `pageDetect`s than spreading `sortMileStonesByClosestDueDate()` in multiple different ones.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by bfred-it** [comment]
To make it truly default, it should touch all the links to **Milestones** that are not explicitly in a different order. These include the Open and Closed links and those added by #366 

screen shot 2017-07-05 at 22 44 12

**Comment by bfred-it** [comment]
Should we get this in?

File: readme.md [hunk 4]
```diff
@@ -135,6 +135,22 @@ GitHub Enterprise is also supported by [authorizing your own domain in the optio
...
+And [lots](extension/content.css) [more...](src/content.js)
+
+
+## Screenshots
+
+### Dashboard
+
+![](media/screenshot-dashboard.png)
+
+### Repo
```
Comment (sindresorhus) [inline readme.md]: This merge conflict seems to have been resolved incorrectly.

Comment (bfred-it) [inline readme.md]: Good catch. I had only ran `git rebase origin/master`

**Pull request merged**